cql3/expr/prepare_expr.cc

```diff
@@ -873,8 +874,26 @@ try_prepare_expression(const expression& expr, data_dictionary::database db, con
         [&] (const binary_operator& binop) -> std::optional<expression> {
             return prepare_binary_operator(binop, db, *schema_opt);
         },
-        [&] (const conjunction&) -> std::optional<expression> {
-            on_internal_error(expr_logger, "conjunctions are not yet reachable via prepare_expression()");
+        [&] (const conjunction& conj) -> std::optional<expression> {
+            std::vector<expression> prepared_children;
+            for (const expression& child : conj.children) {
+                std::optional<expression> prepared_child = try_prepare_expression(child, db, keyspace, schema_opt, receiver);
```

Comment:
You should create a receiver here (of boolean type). This is so that 

`SELECT ? AND ?` works. `? AND ?` doesn't have a receiver (we don't know its type), but we do know the types of `?` and `?` (they must both be booleans).

Please add unit tests for these type inference issues (also for binary_operator).